docs/knowledgebase/learn-substrate/off-chain-workers.md

```diff
@@ -46,3 +46,23 @@ should be implemented to determine what information gets into the chain.
 
 For more information on how to use off-chain workers in your next runtime development project,
 please refer to our [Development Guide](../runtime/off-chain-workers).
+
+# Off-Chain Indexing
+
+Storage in the context of blockchain is mostly about on-chain state. But it is expensive (as 
+it is populated to each node in the network) and not recommended for historical or user-generated 
+data which grow indefinitely over time.
+
+Luckily every Substrate node comes with off-chain database also. One of the main use cases is to be
+used together with off-chain workers mentioned above. In addition, Substrate also includes a feature
+called "off-chain indexing", which allows the runtime to write directly to the off-chain database
+independently from OCWs. Nodes have to opt-in for persistency of this data via 
+`--enable-offchain-indexing` flag when starting up the Substrate node.
+
+Unlike OCWs, which are not executed during initial blockchain synchronization, off-chain indexing is 
+populating the database every time a block is processed, so the data is always consistent and will
+be exactly the same for every node with indexing enabled.
+
+Off-chain database can be read using remote procedure calls (RPC) so it fits the use case of storing
+indefinitely growing data without over-consuming the on-chain storage.
```

Comment:
Maybe let's add the code samples to `Development Guide` as well - right now, since there is no pointers at all it's really hard to go from "I've learned the concept" to "I want to try it out".

Reply:
I would prefer to follow Jimmy's suggestion and keep the code in the Recipes.

Reply:
Thanks input from Tomek and Dan. 

Tomek, I commit that I will write a sample code, either in recipe or the `Runtime` section below on off-chain index before Jan 15th. If we put code out, we likely need a section of code and explain to readers the context. So I need to experiment with the API myself first, and this takes some work and time. Will let you guys review it once it is done. Then we will add a link to the code section here.

So I prefer keeping this section to be conceptual and show the code in another section. Let me know if you are okay with this, @tomusdrw